src/Adapter/EntityMapper.php

```diff
@@ -98,7 +98,12 @@ public function load($id, $id_lang, $entity, $entity_defs, $id_shop, $should_cac
                         if (isset($entity_defs['fields'][$key]['type']) && in_array($entity_defs['fields'][$key]['type'], [
                             \ObjectModel::TYPE_BOOL,
                         ])) {
-                            $entity->{$key} = (string) $value;
+                            if (is_array($value)) {
```

Comment:
Hi @gross-nvs 

I'm not sure to understand the issue here, you try to cast an array for a field that's supposed to be a boolean, how can data for a boolean be an array? Isn't there some inconsistency in the ObjectModel definition here? 🤔

Could you please provide a concrete example of how we can reproduce this error because your PR is not related to any issue, or any reproduced bug and without more context this modification seems doutbful